grpc/server.go

```diff
@@ -68,3 +68,83 @@ func (s *Server) List(ctx context.Context, req *v1.ListRequest) (*v1.ListRespons
 
 	return repoList.ToProto(), nil
 }
+
+// gRPCChunkSender is a zoekt.Sender that sends small chunks of FileMatches to the provided gRPC stream.
+func gRPCChunkSender(ss v1.WebserverService_StreamSearchServer) zoekt.Sender {
+	f := func(r *zoekt.SearchResult) {
+		result := r.ToProto()
+
+		if len(result.GetFiles()) == 0 { // stats-only result, send it immediately
+			_ = ss.Send(result)
+			return
+		}
+
+		// Otherwise, chunk the file matches into multiple responses
+
+		statsSent := false
+		numFilesSent := 0
+
+		sendFunc := func(filesChunk []*v1.FileMatch) error {
+			numFilesSent += len(filesChunk)
+
+			var stats *v1.Stats
+			if !statsSent { // We only send stats back on the first chunk
+				statsSent = true
+				stats = result.GetStats()
+			}
+
+			progress := result.GetProgress()
+
+			if numFilesSent < len(result.GetFiles()) { // more chunks to come
+				progress = &v1.Progress{
+					Priority: result.GetProgress().GetPriority(),
+
+					// We want the client to consume the entire set of chunks - so we manually
+					// patch the MaxPendingPriority to be >= overall priority.
+					MaxPendingPriority: math.Max(
+						result.GetProgress().GetPriority(),
+						result.GetProgress().GetMaxPendingPriority(),
+					),
+				}
+			}
+
+			// We need to ensure that we send the repository URLs and LineFragments for only the repositories
+			// actually mentioned in this chunk of files.
+			repoURLs := make(map[string]string, len(result.GetRepoUrls()))
```

Comment:
cc @keegancsmith 

I implemented repoURL and LineFragment gRPC support since they still seem to be referenced in our application code here: https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+linefragments&patternType=standard&sm=1&groupBy=path

Reply:
Ah, actually I'm now realizing that these results are probably talking about a different type (since linefragments here is a string map, but in sourcegraph it's a struct). I'll run this down more to see if the the string linefragments from Zoekt are somehow used to create that new type .

Reply:
we only use it in one of our aggregation code paths, but never actually use it. So you can remove it. https://github.com/sourcegraph/sourcegraph/pull/55854